### DISCUSSION TITLE
imageScalingMode not working

### DISCUSSION THREAD

**Discussion by schaefsteven** [nix-community/stylix#494] [Q&A] [answered]
I can't seem to get imageScalingMode to work. This is what's in my configuration.nix: 

```
  stylix = {
    enable = true;
    base16Scheme = "${pkgs.base16-schemes}/share/themes/gruvbox-dark-medium.yaml";
    image = ./background-image;
    imageScalingMode = "center";
  };
```
But the wallpaper is using what I'm assuming is "stretch" mode instead. I have 3 monitors: two landscape and one portrait. And on the portrait monitor, the wallpaper is squashed horizontally.

I've tried setting several different values for imageScalingMode, but nothing changes the behavior. 

My displayManager (lightdm) background is using the "fill" mode, but once I log in, the wallpaper changes mode.

I've tried also setting services.xserver.desktopManager.wallpaper.mode, to see if that is overriding it, but nothing I set in that option has any effect either. 

So it seems like the option is either broken, or something else in my system is overriding stylix for this setting. 

NixOS 24.11
lightdm
i3

**Comment by trueNAHO** [comment, marked as answer]
> My displayManager (lightdm) background is using the "fill" mode, but once I log in, the wallpaper changes mode.
>
> [...]
>
> So it seems like the option is either broken, or something else in my system is overriding stylix for this setting.

`stylix.imageScalingMode` is currently only used by the Sway module: c760f63a44a98b2324fdebaee32831b1297172a1.

**Reply by schaefsteven** [reply]
Ahh. Thanks, I did not realize that. 

So is there any way that you are aware of to set the correct scaling on Xorg? 

And if not, is there a way that I can override the stylix wallpaper with something else? Or do I need to wait until issue [#442](https://github.com/danth/stylix/issues/442) is finished?

**Reply by trueNAHO** [reply]
> So is there any way that you are aware of to set the correct scaling on Xorg?

That would require updating the concerned modules to consider `stylix.imageScalingMode`. Whether that is possible depends on the underlying applications.

Alternatively, `stylix.image` could be modified based on `stylix.imageScalingMode`, such that individual applications can be unaware of `stylix.imageScalingMode`. In this case, you might be interested in #477.

> And if not, is there a way that I can override the stylix wallpaper with something else? Or do I need to wait until issue [#442](https://github.com/danth/stylix/issues/442) is finished?

Yes, `stylix.image` is currently not optional.

**Reply by danth** [reply]
> Alternatively, `stylix.image` could be modified based on `stylix.imageScalingMode`, such that individual applications can be unaware of `stylix.imageScalingMode`.

This would require knowing the monitor dimensions in advance, which may not be possible when connecting to an external display or projector for example.

Many applications support this setting, it just hasn't been widely implemented within Stylix yet.
